libweb3jsonrpc/Debug.cpp

```diff
@@ -126,6 +126,38 @@ Json::Value Debug::debug_traceBlockByNumber(int _blockNumber, Json::Value const&
 	return ret;
 }
 
+Json::Value Debug::debug_accountRangeAt(string const& _blockHashOrNumber, int _txIndex, string const& _address, int _maxResults)
+{
+	Json::Value ret(Json::arrayValue);
+
+	if (_txIndex < 0)
+		throw jsonrpc::JsonRpcException("Negative index");
+	if (_maxResults <= 0)
+		throw jsonrpc::JsonRpcException("Nonpositive maxResults");
+
+	try
+	{
+		Block block = m_eth.block(blockHash(_blockHashOrNumber));
+
+		unsigned const i = ((unsigned)_txIndex < block.pending().size()) ? (unsigned)_txIndex : block.pending().size();
```

Comment:
check Debug::debug_storageRangeAt  below this method

Reply:
I mean should I fix that part too I suppose I should.

Reply:
You can but that's not required.
Here I'd do:
```
size_t const i = std::min(static_cast<size_t>(_txIndex), block.pending().size());
```

But I'm not sure this is good logic. Shouldn't we report error in case the transaction index is wrong? And txIndex == block.pending().size() seems to be also wrong.